### ISSUE TITLE
I should be able to see the underlying Terraform of a template in the UI. 

### ISSUE THREAD

**Issue by sharkymark** [coder/coder#2204]
I see absolutely nothing and have to track down the creator to see the Terraform.

Or create a workspace and read the logs for the Terraform run.

**Comment by ketang** [comment]
Qualified yes.  Visibility should be limited and controlled by the template owner.  Lack of global visibility is a security feature.

**Comment by bpmct** [comment]
Suggestion: Add an item to the template metadata called "source-url" and render that in the template Ui. Reasoning
- some templates may include multiple Terraform files and reference other assets
- this also allows developers to potentially submit PRs to modify the template, if needed.

Reminds me of the "built from template" in v1.



```diff
# e.g README.md for aws-linux template

---
name: Develop in Linux on AWS EC2
description: Get started with Linux development on AWS EC2.
tags: [cloud, aws]
+ source-url: https://github.myorg.com/coder-templates/tree/main/aws-linux
---

# aws-linux

To get started, run `coder templates init`. When prompted, select this template.
Follow the on-screen instructions to proceed.
```
![ss](https://user-images.githubusercontent.com/22407953/172989048-fe2d2f5e-a6db-4ef2-bd48-b42069a75cf8.png)

**Comment by ketang** [comment]
Would that then rely on their source control's access mechanisms?   That works.

One thing that will quickly follow this is a customer request to support pulling the scripts out of source control and automatically updating them.

**Comment by ketang** [comment]
#2240 for the latter.

**Comment by tjcran** [comment]
> #2240 for the latter.

#2240  is now closed as a duplicate of #2958

**Comment by kylecarbs** [comment]
I want to await feedback before we do this. It's a bit weird to just display the Terraform source code in the UI, but I understand why people might want it.
